pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -500,6 +510,34 @@ func (gce *GCECloud) ensureTargetPoolAndHealthCheck(tpExists, tpNeedsRecreation
 			glog.Infof("ensureTargetPoolAndHealthCheck(%s): Updated target pool (with %d hosts).", lbRefStr, len(hosts)-maxTargetPoolCreateInstances)
 		}
 	} else if tpExists {
+
+		if hcToDelete != nil {
+			if hc, err := gce.GetHttpHealthCheck(hcToDelete.Name); err != nil {
+				if !isHTTPErrorCode(err, http.StatusNotFound) {
+					return fmt.Errorf("error getting HTTP health check for load balancer (%s): %v", lbRefStr, err)
+				}
+			} else {
+				healthchecks := []*compute.HealthCheckReference{{HealthCheck: hc.SelfLink}}
+				if err := gce.RemoveHealthChecksFromTargetPool(loadBalancerName, gce.region, healthchecks); err != nil {
```

Comment:
What happen if we mistakenly pass in healthcheck that isn't referred by the targetpool?

Reply:
Check added.